src/shared/modules/auth/common.js

```diff
@@ -0,0 +1,217 @@
+  AUTH_STORAGE_URL_SEARCH_PARAMS,
+  REDIRECT_URI
+} from './constants'
+import { addSearchParamsInBrowserHistory, authLog, authDebug } from './helpers'
+import {
+  defaultTokenTypeAuthentication,
+  defaultTokenTypePrincipal,
+  mandatoryKeysForSSOProviderParams,
+  mandatoryKeysForSSOProviders
+} from './settings'
+
+export const getInitialisationParameters = () => {
+  const urlSearchParams = window.location.search
+  const urlHashParamsAsSearchParams = '?' + window.location.hash.substring(1)
+
+  const initParams = {}
+  new URLSearchParams(urlSearchParams).forEach((value, key) => {
+    initParams[key] = value
+  })
+  new URLSearchParams(urlHashParamsAsSearchParams).forEach((value, key) => {
+    initParams[key] = value
+  })
+
+  return initParams
+}
+
+export const checkAndMergeSSOProviders = (
+  discoveredSSOProviders,
+  isLocalhostOrigin
+) => {
+  if (!discoveredSSOProviders || !discoveredSSOProviders.length) {
+    authLog('Invalid discovered SSO providers')
+    return
+  }
+
+  let currentSSOProviders =
+    JSON.parse(window.sessionStorage.getItem(AUTH_STORAGE_SSO_PROVIDERS)) || []
+  if (!Array.isArray(currentSSOProviders)) {
+    authLog(
+      'Found SSO providers in storage are defect, consider clearing the SSO provider state and reload',
+      'warn'
+    )
+    currentSSOProviders = []
+  }
+
+  discoveredSSOProviders.forEach(provider => {
+    if (!provider) return
+    if (
+      !mandatoryKeysForSSOProviders.every(key => provider.hasOwnProperty(key))
+    ) {
+      authLog(
+        `Dropping invalid discovered SSO provider with id: "${provider.id}", missing key`
+      )
+      return
+    }
+    if (
+      !mandatoryKeysForSSOProviderParams.every(key =>
+        provider?.params?.hasOwnProperty(key)
+      )
+    ) {
+      authLog(
+        `Dropping invalid discovered SSO provider with id: "${provider.id}", missing params key`
+      )
+      return
+    }
+    if (
+      currentSSOProviders.find(crntProvider => crntProvider.id === provider.id)
+    ) {
+      if (isLocalhostOrigin) {
+        const idx = currentSSOProviders.findIndex(
+          crntProvider => crntProvider.id === provider.id
+        )
+        currentSSOProviders.splice(idx, 1)
+        authLog(`Updating SSO provider with id: "${provider.id}"`)
+      } else {
+        authLog(
+          `Not accepting discovered SSO provider with id: "${provider.id}", id exists already`
+        )
+        return
+      }
+    }
+    currentSSOProviders.push(provider)
+  })
+
+  window.sessionStorage.setItem(
+    AUTH_STORAGE_SSO_PROVIDERS,
+    JSON.stringify(currentSSOProviders)
+  )
+  authLog('Checked and merged SSO providers')
+}
+
+export const getSSOProvidersFromStorage = () => {
+  const ssoProviders = JSON.parse(
+    window.sessionStorage.getItem(AUTH_STORAGE_SSO_PROVIDERS)
+  )
+  if (!ssoProviders || !ssoProviders.length) {
+    authLog('No SSO providers in (local) storage found')
+    return []
+  }
+  if (!window.isSecureContext) {
```

Comment:
we should have a "big hammer" version of this check much earlier imo...

Reply:
Moved this to be before the redirect as well